samples/wifi/radio_test/CMakeLists.txt

```diff
@@ -6,6 +6,7 @@
 
 cmake_minimum_required(VERSION 3.20.0)
 
+set(ZEPHYR_EXTRA_MODULES ${CMAKE_CURRENT_LIST_DIR})
```

Comment:
It is an abuse of the module system to add a "module" which in fact is the same git repository. It looks like you can just use add_subdirectory instead, no?

Reply:
This was based on the sample https://github.com/nrfconnect/sdk-nrf/blob/main/samples/nrf5340/multicore/CMakeLists.txt#L9

Reply:
> This was based on the sample https://github.com/nrfconnect/sdk-nrf/blob/main/samples/nrf5340/multicore/CMakeLists.txt#L9

That does not make this correct.

> It looks like you can just use add_subdirectory instead, no?

Please address this.

Reply:
> This was based on the sample https://github.com/nrfconnect/sdk-nrf/blob/main/samples/nrf5340/multicore/CMakeLists.txt#L9

That does not make this correct.

> It looks like you can just use add_subdirectory instead, no?

Please address this.

Reply:
> > This was based on the sample https://github.com/nrfconnect/sdk-nrf/blob/main/samples/nrf5340/multicore/CMakeLists.txt#L9
> 
> That does not make this correct.
> 
Sure, I was just pointing to the original source, we did various experiments and none of them worked, so, had to go with that verbatim. You are right that https://docs.zephyrproject.org/3.1.0/develop/modules.html#integrate-modules-in-zephyr-build-system suggests to use that, but this is still external to Zephyr as it is part of `sdk-nrf`.

> > It looks like you can just use add_subdirectory instead, no?
> 
> Please address this.

I tried this, but it doesn't work, as there is no `CMakeLists.txt` in the previous directories where the `add_subdirectory` should be added, and we cannot the `CMakeLists.txt` in the all the previous directories as we only want to include the sample of interest, so, I am not really sure how to achieve this with `add_subdirectory`. (P.S. This might be why Zephyr external module approach might have been taken even for `sdk-nrf` samples).

Reply:
OK, thanks for the response. In that case I think that the way to go might be to accept this as technical debt that will need to be converted to use sysbuild. cc @tejlmand @nordicjm -- OK with the two of you?

Reply:
>  cc @tejlmand @nordicjm -- OK with the two of you?

ping; please confirm this is OK; I am blocking the PR until we can decide whether this is tech debt we can accept and have a plan for managing, or whether this is something we need to find another approach for

Reply:
@hakonfam is this the correct method for including a child image?

Reply:
The "extra module trick" was how @tejlmand suggested that this problem (adding child images from sample cmakelists) was solved back in the "pre sysbuild" era. I don't have any experience with sysbuild, so I don't know if  the correct way of doing this has changed after the introduction of sysbuild.

Reply:
> The "extra module trick" was how @tejlmand suggested that this problem (adding child images from sample cmakelists) was solved back in the "pre sysbuild" era.

Please note that this was a show case how a user may have a skeleton available to see how they can create a child image.
This is not the same as we should have dedicated child images inside our own samples.

If users are using https://github.com/nrfconnect/sdk-nrf/tree/main/samples/nrf5340/multicore as a reference point, the best solution for them is to actually place the `add_child_image()` call in dedicated Zephyr module code, but that's impractical to show in a sample. Guess the sample doc / CMake code can be improved to reflect this.

For other NCS samples,m such as this PR, then a sample should be a clean project that customers may copy out and adjust according to their needs.

This PR uses another sample in NCS as a child image: https://github.com/nrfconnect/sdk-nrf/blob/2405a3d702ed74cfd2c65f36859be188b69f26dc/samples/wifi/radio_test/aci/CMakeLists.txt#L9
and therefore this child image should be generally available like other child images available in NCS, such as `hci_rpmsg`, `spm`, `remote_shell`, etc.
That will also allow for corresponding Kconfig setting to enable the radio test sample (and to safeguard it properly).